### DISCUSSION TITLE
The speed of uploading images on a self hosted outline is very slow, and it is even possible that the upload may fail.

### DISCUSSION THREAD

**Discussion by shanyan-wcx** [outline/outline#6164] [Self Hosting] [answered]
The speed of uploading images on a self hosted outline is very slow, and it is even possible that the upload may fail.
This is just a 2M sized image.

**To Reproduce**
1. Upload an image.
2. Long wait.
3. It is likely that the upload failed.

**Expected behavior**
Successfully uploaded quickly.

**Screenshots**
https://github.com/outline/outline/assets/58252651/30b0c6b2-8dc9-48ee-8a34-13e3e6b640d4

**Outline (please complete the following information):**
- Install: [self hosted]
- Version: [v0.73.1]

**Desktop (please complete the following information):**
 - OS: [windows11]
 - Browser [edge]
 - Version [119.0.2151.58]

**Comment by tommoor** [comment]
The uploads is performed directly from the browser to file storage, what file storage are you using?

**Reply by shanyan-wcx** [reply]
local

**Comment by shanyan-wcx** [comment, marked as answer]
It seems that this issue occurred because I used CloudflaredTunnel. Because I don't have a public IPv4 address, I used CloudflaredTunnel to expose the outline to the public network. But I don't understand why this is happening.

**Comment by mercgarage** [comment]
Same issue, it was working fine, but today noticed it takes ages to upload even a small kb file. It is hosted in container station.

**Comment by mercgarage** [comment]
I think found the issue. It is geting slow when we add lots of info in a doc, contain many tables and attachments. For new doc the pic or any attachment is simple copy paste that much fast. When doc get bigger then it start slowing down.

**Comment by tajchert** [comment]
This is a real issue (also similar setup, local storage, CF tunnel for selfhosting) and event clean Outline instance with single document takes around 2 minutes to upload 2MB image.

**Reply by tommoor** [reply]
No idea about CF tunnel, you'd have to take it up with that provider – seems to be the common thread
